altair/utils/save.py

```diff
@@ -15,6 +16,48 @@ def write_file_or_filename(fp, content, mode="w"):
         fp.write(content)
 
 
+def set_inspect_format_argument(format, fp, inline):
+    """Inspect the format argument in the save function"""
+    if format is None:
+        if isinstance(fp, str):
+            format = fp.split(".")[-1]
+        elif isinstance(fp, pathlib.PurePath):
+            format = fp.suffix.lstrip(".")
```

Comment:
We could associate the file endings `.vg` and `.vg.json` with `vega` and `.vl`, `.vl.json` with `vega-lite` so the files can be opened in JupyterLab https://jupyterlab.readthedocs.io/en/2.2.x/user/file_formats.html#vega-vega-lite

Reply:
Opend https://github.com/altair-viz/altair/issues/2852 to track this